pinot-query-planner/src/main/java/org/apache/pinot/query/routing/StageMetadata.java

```diff
@@ -29,10 +29,13 @@
  */
 public class StageMetadata {
   private final List<WorkerMetadata> _workerMetadataList;
+  private final Map<Integer, List<MailboxInfo>> _mailBoxInfosMap;
   private final Map<String, String> _customProperties;
 
-  public StageMetadata(List<WorkerMetadata> workerMetadataList, Map<String, String> customProperties) {
+  public StageMetadata(List<WorkerMetadata> workerMetadataList, Map<Integer, List<MailboxInfo>> mailBoxInfosMap,
```

Comment:
MailboxInfo should be on WorkerMetadata level

Reply:
yes, changed to list, the ordering has to be kept.